docs/source/api_references.rst

```diff
@@ -65,14 +65,27 @@ Callbacks API
 .. autosummary::
     :toctree: api
     :nosignatures:
+    :template: classtemplate.rst
 
-    base
-    early_stopping
-    gpu_stats_monitor
-    gradient_accumulation_scheduler
-    lr_monitor
-    model_checkpoint
-    progress
+    Callback
+    DeviceStatsMonitor
+    EarlyStopping
+    BaseFinetuning
+    BackboneFinetuning
+    GPUStatsMonitor
+    GradientAccumulationScheduler
+    LambdaCallback
+    LearningRateMonitor
+    ModelCheckpoint
+    BasePredictionWriter
+    ModelPruning
+    QuantizationAwareTraining
+    RichModelSummary
```

Comment:
ModelSummary is missing. Maybe use this list just to ensure all are there: https://pytorch-lightning.readthedocs.io/en/latest/extensions/callbacks.html#built-in-callbacks

Reply:
Good catch, updated!

Reply:
Why do we have these in multiple spots? Doesn't that keep the risk of these falling out of sync?